syz-manager/manager.go

```diff
@@ -345,16 +372,16 @@ func (mgr *Manager) vmLoop() {
 			}
 			Logf(1, "loop: add to repro queue '%v'", crash.desc)
 			reproducing[crash.desc] = true
-			reproQueue = append(reproQueue, crash)
+			mgr.reproQueue = append(mgr.reproQueue, crash)
 		}
 
-		Logf(1, "loop: phase=%v shutdown=%v instances=%v/%v %+v repro: pending=%v reproducing=%v queued=%v",
+		Logf(1, "loop: phase=%v shutdown=%v instances=%v/%v %+v repro: pending=%v reproducing=%v queued=%v hub=%v",
 			phase, shutdown == nil, len(instances), vmCount, instances,
-			len(pendingRepro), len(reproducing), len(reproQueue))
+			len(pendingRepro), len(reproducing), len(mgr.reproQueue), len(mgr.hubReproQueue))
 
 		canRepro := func() bool {
-			return phase >= phaseTriagedHub &&
-				len(reproQueue) != 0 && reproInstances+instancesPerRepro <= vmCount
+			return phase >= phaseTriagedCorpus && reproInstances+instancesPerRepro <= vmCount &&
+				(len(mgr.hubReproQueue) != 0 || len(mgr.reproQueue) != 0)
```

Comment:
You work with hubReproQueue like it's an array. Channels are not ordinary data structures. Its size changes asynchronously. They should be used for communication. Querying asynchronously changing size is this kind of logic predicate looks wrong. You also still have the problem of vmLoop not reacting on sends to hubReproQueue. Which will lead to delayed reproduction and hangs in hubSync.
You need to receive from hubReproQueue in the select below and put the received crash to reproQueue (which should be local to vmLoop). This makes vmLoop react on new crashes instantly, and also does not require any complication of the existing predicates and logic in vmLoop. And also should resolve deadlocks that you see.